mantra/representations/simplicial_connectivity.py

```diff
@@ -0,0 +1,300 @@
+from typing import Dict, List
+from networkx import incidence_matrix
+import torch
+import numpy as np
+from scipy.sparse import csr_matrix
+import scipy
+
+import torch_geometric.transforms as T
+from torch_geometric.data import Data
+
+from mantra.representations.simplex_trie import SimplexTrie, Simplex
+
+
+class AddSimplexTrie(T.BaseTransform):
+    """Add simplex trie to object.
+
+    Takes in a `Data` object and computes a `SimplexTrie` for it.
+    Then it adds it to it's storage to save up on computation
+    of connectivity matrices.
+
+
+    """
+
+    def forward(self, data: Data):
+        assert "triangulation" in data
+        simplex_trie = SimplexTrie()
+        for t in data.triangulation:
+            t_ = frozenset(t)
+            simplex_trie.insert(sorted(t_))
+        data.simplex_trie = simplex_trie
+        return data
+
+
+class ConnectivitySCTransform(T.BaseTransform):
+    """Base class for connectivity transforms.
+
+    Parent class for implementing a transform that adds a
+    connectivity matrix for simplices. This transform takes a `Data` object with a `triangulation`
+    field and represents this triangulation by the canonical
+    representation of it's neighborhood relationships in the
+    PyG edge-index format.
+
+    """
+
+    def __init__(self, signed: bool, connectivity_name: str, index: bool):
+        self.signed = signed
+        self.connectivity_name = connectivity_name
+        self.index = index
+
+    def skeleton(self, simplex_trie: SimplexTrie, rank: int) -> List[Simplex]:
+        """Compute the r-skeleton of a simplex.
+
+        Parameters
+        ----------
+        simplex_trie: SimplexTrie
+            The datastructure containing a simplex.
+        rank: int
+            Rank r of the skeleton.
+        Returns:
+            List[Simplex]
+
+        """
+        return sorted(node.simplex for node in simplex_trie.skeleton(rank))
+
+    def generate_matrix(
```

Comment:
Let's make this abstract to ensure that it's clear that this needs implementation.